nibabies/workflows/bold/resampling.py

```diff
@@ -117,63 +156,518 @@ def select_target(subject_id, space):
+            (cov_ribbon_mean, cov_norm, [("out_stat", "operand_value")]),
+            (cov_volume, cov_norm, [("out_file", "in_file")]),
+            (cov_norm, cov_norm_modulate, [("out_file", "in_file")]),
+            (cov_ribbon_norm_smooth, cov_norm_modulate, [("out_file", "operand_file")]),
+            (cov_norm_modulate, cov_norm_modulate_ribbon, [("out_file", "in_file")]),
+            (ribbon_boldsrc_xfm, cov_norm_modulate_ribbon, [("output_image", "mask_file")]),
+            (cov_norm_modulate_ribbon, mod_ribbon_mean, [("out_file", "in_file")]),
+            (cov_norm_modulate_ribbon, mod_ribbon_std, [("out_file", "in_file")]),
+            (mod_ribbon_mean, merge_mod_ribbon_stats, [("out_stat", "in1")]),
+            (mod_ribbon_std, merge_mod_ribbon_stats, [("out_stat", "in2")]),
+            (merge_mod_ribbon_stats, upper_thr_val, [("out", "in_stats")]),
+            (merge_mod_ribbon_stats, lower_thr_val, [("out", "in_stats")]),
+            (mean_volume, bin_mean_volume, [("out_file", "in_file")]),
+            (upper_thr_val, goodvoxels_thr, [("upper_thresh", "thresh")]),
+            (cov_norm_modulate, goodvoxels_thr, [("out_file", "in_file")]),
+            (bin_mean_volume, merge_goodvoxels_operands, [("out_file", "in1")]),
+            (goodvoxels_thr, goodvoxels_mask, [("out_file", "in_file")]),
+            (merge_goodvoxels_operands, goodvoxels_mask, [("out", "operand_files")]),
+        ]
+    )
+
+    return workflow, goodvoxels_mask, ribbon_boldsrc_xfm
+
+
+def project_vol_to_surf(workflow, mem_gb, rename_src, goodvoxels_mask, ribbon_boldsrc_xfm):
+    """
+    Parameters
+    ----------
+    workflow : :obj:
+    mem_gb : :obj:`float`
+        Size of BOLD file in GB
+    rename_src: _type_, _description_
+    goodvoxels_mask: _type_, _description_
+    ribbon_boldsrc_xfm: _type_, _description_
+
+    :return: _type_, _description_
+    """
+    goodvoxels_ribbon_mask = pe.Node(
+        fsl.ApplyMask(),
+        name_source=['in_file'],
+        keep_extension=True,
+        name="goodvoxels_ribbon_mask",
+        mem_gb=DEFAULT_MEMORY_MIN_GB,
+    )
+
+    apply_goodvoxels_ribbon_mask = pe.Node(
+        fsl.ApplyMask(),
+        name_source=['in_file'],
+        keep_extension=True,
+        name="apply_goodvoxels_ribbon_mask",
+        mem_gb=mem_gb * 3,
+    )
+
+    # apply goodvoxels ribbon mask to bold
+    workflow.connect(
+        [
+            (goodvoxels_mask, goodvoxels_ribbon_mask, [("out_file", "in_file")]),
+            (ribbon_boldsrc_xfm, goodvoxels_ribbon_mask, [("output_image", "mask_file")]),
+            (goodvoxels_ribbon_mask, apply_goodvoxels_ribbon_mask, [("out_file", "mask_file")]),
+            (rename_src, apply_goodvoxels_ribbon_mask, [("out_file", "in_file")]),
+        ]
+    )
+
+    return workflow, apply_goodvoxels_ribbon_mask
+
+
+def apply_med_surf_nans_if(
+    medial_surface_nan, workflow, inputnode, sampler, update_metadata, medial_nans
```

Comment:
We generally stay away from passing workflows / inputs / nodes as parameters, as that greatly complicates the logic / testing.

These connections can be applied within the `init_` workflow function